sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java

```diff
@@ -255,27 +256,19 @@ public Resource build() {
     }
 
     private Resource.ReservationInfo getRefinedReservationInfo(String role, String resId) {
-        return Resource.ReservationInfo.newBuilder()
-                .setPrincipal(principal.get())
+        Resource.ReservationInfo.Builder reservationBuilder = Resource.ReservationInfo.newBuilder()
                 .setRole(role)
                 .setType(Resource.ReservationInfo.Type.DYNAMIC)
-                .setLabels(
-                        Protos.Labels.newBuilder()
-                                .addLabels(Protos.Label.newBuilder()
-                                        .setKey(MesosResource.RESOURCE_ID_KEY)
-                                        .setValue(resId)))
-                .build();
+                .setPrincipal(principal.get());
+        AuxLabelAccess.setResourceId(reservationBuilder, resId);
```

Comment:
Labels are somewhat special as they're stored in ZK (generally via TaskInfos, sometimes elsewhere), and reused by schedulers across releases. Moving labels into the special `taskdata` package at least makes it clearer where the boundary is between stuff that can change and stuff that needs a migration strategy.